dataactvalidator/config/sqlrules/b9_award_financial.sql

```diff
@@ -22,16 +24,18 @@ SELECT
 FROM award_financial_b9_{0} AS af
      INNER JOIN submission AS sub
         ON af.submission_id = sub.submission_id
-WHERE (af.program_activity_code <> '0000'
-        OR UPPER(af.program_activity_name) <> 'UNKNOWN/OTHER')
-    AND (sub.reporting_fiscal_year, sub.reporting_fiscal_period) NOT IN (('2017', 6), ('2017', 9))
-    AND NOT EXISTS (
-        SELECT 1
-        FROM program_activity AS pa
-        WHERE af.agency_identifier = pa.agency_id
-            AND af.main_account_code = pa.account_number
-            AND UPPER(COALESCE(af.program_activity_name, '')) = UPPER(pa.program_activity_name)
-            AND UPPER(COALESCE(af.program_activity_code, '')) = UPPER(pa.program_activity_code)
-            AND pa.fiscal_year_period = 'FY' || RIGHT(CAST(sub.reporting_fiscal_year AS CHAR(4)), 2) || 'P' || LPAD(sub.reporting_fiscal_period::text, 2, '0')
-
+WHERE
+    (sub.reporting_fiscal_year, sub.reporting_fiscal_period) NOT IN (('2017', 6), ('2017', 9))
+    AND (af.program_activity_code IS NOT NULL OR af.program_activity_name IS NOT NULL)
+    AND (
+        (af.program_activity_code = '0000' AND UPPER(af.program_activity_name) = 'UNKNOWN/OTHER')
```

Comment:
If you're already checking that it's not null, why make that a separate `AND` instead of just running a `COALESCE` here? That just seems like an extra check.